BE/src/main/java/louie/dong/airbnb/accommodation/AccommodationService.java

```diff
@@ -0,0 +1,116 @@
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import louie.dong.airbnb.accommodation.dto.AccommodationDetailPriceRequest;
+import louie.dong.airbnb.accommodation.dto.AccommodationDetailPriceResponse;
+import louie.dong.airbnb.accommodation.dto.AccommodationDetailResponse;
+import louie.dong.airbnb.accommodation.dto.AccommodationPriceResponse;
+import louie.dong.airbnb.accommodation.dto.AccommodationResponse;
+import louie.dong.airbnb.accommodation.dto.AccommodationSearchRequest;
+import louie.dong.airbnb.accommodation.dto.AccommodationSearchResponse;
+import louie.dong.airbnb.wishlist.WishRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class AccommodationService {
+
+	private final AccommodationRepository accommodationRepository;
+	private final WishRepository wishRepository;
+
+	public AccommodationPriceResponse findPrices(String country) {
+		List<Integer> prices = accommodationRepository.findByAccommodationPrices(country);
+		return new AccommodationPriceResponse(calculateAverage(prices), prices);
+	}
+
+	public AccommodationDetailResponse findById(Long id) {
+		Accommodation accommodation = getAccommodation(id);
+		boolean wish = wishRepository.existsByAccommodationId(id);
+		return new AccommodationDetailResponse(accommodation, wish);
+	}
+
+	public AccommodationDetailPriceResponse findDetailPrice(Long id,
+		AccommodationDetailPriceRequest accommodationDetailPriceRequest) {
+		Accommodation accommodation = getAccommodation(id);
+		LocalDate checkIn = accommodationDetailPriceRequest.getCheckIn();
+		LocalDate checkOut = accommodationDetailPriceRequest.getCheckOut();
+
+		int date = (int) checkIn.until(checkOut, ChronoUnit.DAYS);
+		int totalPrice = accommodation.getPrice() * date;
+		double discountRate = getDiscountRate(date) * 0.01;
+		int discountPrice = (int) (totalPrice * discountRate);
+		int finalPrice = totalPrice - discountPrice + accommodation.getCleaningFee()
+			+ accommodation.getServiceFee() + accommodation.getAccommodationFee();
+
+		return new AccommodationDetailPriceResponse(accommodation.getPrice(), date, totalPrice,
+			WEEKLY.getDiscountRate(), discountPrice, accommodation.getCleaningFee(),
+			accommodation.getServiceFee(), accommodation.getAccommodationFee(), finalPrice);
+	}
+
+	public AccommodationSearchResponse findAccommodations(
+		AccommodationSearchRequest accommodationSearchRequest) {
+		LocalDate checkIn = accommodationSearchRequest.getCheckIn();
+		LocalDate checkOut = accommodationSearchRequest.getCheckOut();
+
+		List<Accommodation> accommodations = accommodationRepository.searchAccommodations(
+			accommodationSearchRequest.getCountry(),
+			LocalDateTime.of(checkIn, LocalTime.of(0, 0)),
+			LocalDateTime.of(checkOut, LocalTime.of(0, 0)),
```

Comment:
checkin, checkOut 데이터를 LocalDateTime으로 요청받으면 어떨까요?
또는 dto 내부에서 dateTime으로 변환된 데이터를 반환하는건 어떨까요?
```suggestion
			accommodationSearchRequest.getCheckInDateTime(),
			accommodationSearchRequest.getCheckOutDateTime(),
```

Reply:
`LocalDateTime`으로 받는 게 좋은 것 같다는 생각이 들지만 그렇게되면 API 스펙이 바뀌는 것 같아 `dto` 내부에서 데이터를 변환하여 반환하도록 변경하겠습니다!!